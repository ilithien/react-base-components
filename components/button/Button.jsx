import React from "react";
import PropTypes from "prop-types";
import "./Button.css";

const Button = ({ onClick, children }) => (
  <button className="react-base__button" onClick={onClick}>
    {children}
  </button>
);

export default Button;
