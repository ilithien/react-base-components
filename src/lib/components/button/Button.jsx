import React from "react";
import PropTypes from "prop-types";
import "./Button.css";

const Button = ({ onClick, children }) => (
  <button className="react-base__button" onClick={onClick}>
    {children}
  </button>
);

Button.propTypes = {
  children: PropTypes.oneOf([
    PropTypes.element,
    PropTypes.string,
    PropTypes.number,
    PropTypes.node
  ]),
  onClick: PropTypes.func
};

export default Button;
