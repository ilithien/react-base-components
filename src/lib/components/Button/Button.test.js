import React from "react";
import renderer from "react-test-renderer";
import Button from "./Button";

describe("Testing Button component", () => {
  it("matches snapshot", () => {
    const snap = renderer.create(<Button />);

    const tree = snap.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
