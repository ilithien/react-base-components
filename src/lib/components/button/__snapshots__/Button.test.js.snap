// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Button component matches snapshot 1`] = `
<button
  className="react-base__button"
  onClick={undefined}
/>
`;
